pkg/controller/injector/k8s_binding_injector.go

```diff
@@ -64,15 +63,10 @@ func (b *k8sBindingInjector) Inject(binding *servicecatalog.Binding, cred *broke
 		Data: make(map[string][]byte),
 	}
 
-	// For each item in the cred just serialize its value into JSON and
-	// save it in the secret
+	// For each item in the cred, just convert it to the bytes for a string
+	// representation of the value
 	for k, v := range *cred {
-		data, err := json.Marshal(v)
-		if err != nil {
-			return fmt.Errorf("Unable to marshal credential value %q: %s",
-				k, err)
-		}
-		secret.Data[k] = data
+		secret.Data[k] = []byte(fmt.Sprintf("%v", v))
```

Comment:
What I mean is we should add tests for creds that come back like:
```
"user1": "password",
"user2": { "key" : "value" },
"user3": [ "value" ],
"user4": { "key" : { "key2":"value" }}
...
```
Just the various ways the json might appear.  In particular to make sure the nesting of types is printed correctly when using %v.  Regardless of whether any potential bug is in golang or not, we'd be the ones responsible for finding a work-around.